lib/pry/config/lazy.rb

```diff
@@ -0,0 +1,26 @@
+module Pry::Config::Lazy
+  LAZY_KEYS = {}
+  LAZY_KEYS.default_proc = lambda {|h,k| h[k] = [] }
+
+  module ExtendModule
+    def lazy_implement(method_name_to_func)
+      method_name_to_func.each do |method_name, func|
+        define_method(method_name) do
+          if method_name_to_func[method_name].equal?(func)
+            LAZY_KEYS[self.class] |= method_name_to_func.keys
```

Comment:
What's `self.class`? Will it always be the same? Why not a simple Array for `LAZY_KEYS`?